server/aesdsocket.c

```diff
@@ -0,0 +1,266 @@
+        {
+            syslog(LOG_ERR, "Error setting socket options");
+            retval = -1;
+            break;
+        }
+
+        if (bind(sock_fd, p->ai_addr, p->ai_addrlen) != 0)
+        {
+            syslog(LOG_ERR, "Error binding");
+            retval = -1;
+            break;
+        }
+    }
+
+    freeaddrinfo(serv_info);
+
+    // Check for daemon, initializing if option was specified
+    if (run_daemon)
+    {
+        pid_t pid = fork();
+
+        if (pid < 0)
+        {
+            close(sock_fd);
+            exit(EXIT_FAILURE);
+        }
+        else if (pid > 0)
+        {
+            exit(EXIT_SUCCESS);
+        }
+
+        // Create new session
+        if (setsid() == -1)
+        {
+            close(sock_fd);
+            exit(EXIT_FAILURE);
+        }
+
+        // Change working directory
+        if (chdir("/"))
+        {
+            close(sock_fd);
+            exit(EXIT_FAILURE);
+        }
+
+        // Close file descriptors
+        close(STDIN_FILENO);
+        close(STDOUT_FILENO);
+        close(STDERR_FILENO);
+
+        // Redirect stdin/out/err to /dev/null
+        open("/dev/null", O_RDWR);
+        dup(0);
+        dup(0);
+    }
+
+    // listen for connection
+    if (listen(sock_fd, 5) != 0)
+    {
+        syslog(LOG_ERR, "Error listening");
+        retval = -1;
+    }
+    
+    // Accept connections until SIGINT or SIGTERM received
+    while (retval != -1)
+    {
+        // accept connection
+        socklen_t client_addr_size = sizeof(client_addr);
+        client_fd = accept(sock_fd, (struct sockaddr *)&client_addr, &client_addr_size);
+        if (client_fd == -1)
+        {
+            syslog(LOG_ERR, "Issue accepting connection");
+            continue;
+        }
+
+        // log message to syslog when client connects
+        char ip_addr[INET6_ADDRSTRLEN];
+        inet_ntop(client_addr.ss_family,
+                  get_in_addr((struct sockaddr *)&client_addr),
+                  ip_addr, sizeof(ip_addr));
+
+        syslog(LOG_USER, "Accepted connection from %s", ip_addr);
+
+        // Create /var/tmp/aesdsocketdata & append data received
+        char buf[256] = {0};
+        int  bytes_recv;
+        bool more_bytes_to_read = true;
+        fp = fopen(LOG_FILE, "a+");
+        while (more_bytes_to_read)
+        {
+            // Receive data over connection
+            bytes_recv = recv(client_fd, buf, 256, 0);
+            if (bytes_recv == -1)
+            {
+                syslog(LOG_ERR, "Error receiving data");
+                more_bytes_to_read = false;
+                retval = -1;
+                continue;
+            }
+            else if (bytes_recv == 0)
+            {
+                more_bytes_to_read = false;
+            }
+            
+            syslog(LOG_INFO, "Received %d bytes", bytes_recv);
+
+            // Each packet is complete when newline character is received
+            char * new_line_found = memchr(buf, '\n', 256);
```

Comment:
@kbiggs don't hard code this length in multiple places, use a #define or even better a const unsigned int.